contracts/reputationMiningCycle/ReputationMiningCycle.sol

```diff
@@ -547,7 +547,9 @@ contract ReputationMiningCycle is ReputationMiningCycleCommon {
   function startMemberOfPair(uint256 _roundNumber, uint256 _index) internal {
     Submission storage submission = reputationHashSubmissions[disputeRounds[_roundNumber][_index].firstSubmitter];
     disputeRounds[_roundNumber][_index].lastResponseTimestamp = block.timestamp;
-    disputeRounds[_roundNumber][_index].upperBound = submission.jrhNLeaves - 1;
+    if (submission.jrhNLeaves != 0) {
```

Comment:
Was this meant to underflow pre Solidity 8?

Reply:
I _believe_ this underflowed in a test.

Reply:
[Yup](https://app.circleci.com/pipelines/github/JoinColony/colonyNetwork/5159/workflows/c44b481c-cb68-499c-8343-15f6e287cc68/jobs/29203)

Reply:
So I think this comes up if a dispute starts when one member of the pair hasn't yet confirmed their JRH. This previously set the value of upper bound to UINT256_MAX, but now sets it to zero. We use the [https://github.com/JoinColony/colonyNetwork/blob/98c38b0ed480fbb0165fe6a175dbb5b4805575de/contracts/reputationMiningCycle/ReputationMiningCycleRespond.sol#L40](equality of the upper and lower bound) to indicate that the search has finished, so I think it's probably better if we restore the original behavior in this case? I could even believe this underflow was someone (quite possibly me) thinking they were being clever. 

If you agree with this analysis and do this while I'm away, rebase that commit so it's the penultimate commit of the PR, and the last one remains the version bumps, and both have builds on circle. It's a faff, but I want to be careful with this PR :smile:

Reply:
I do agree with the analysis, but will wait for you to make the changes to your liking.